bin/filter-reads-alignments.py

```diff
@@ -0,0 +1,205 @@
+              'zero to only keep reads with no matches (alignments).'))
+
+    parser.add_argument(
+        '--scoreCutoff', type=float, default=None,
+        help=('A float score. Matches with scores worse than this will be '
+              'ignored.'))
+
+    parser.add_argument(
+        '--maxHspsPerHit', type=int, default=None,
+        help='A numeric max number of HSPs to show for each hit on hitId.')
+
+    parser.add_argument(
+        '--whitelist', nargs='+', default=None, action='append',
+        help='sequence titles that should be whitelisted')
+
+    parser.add_argument(
+        '--blacklist', nargs='+', default=None, action='append',
+        help='sequence titles that should be blacklisted')
+
+    parser.add_argument(
+        '--titleRegex', default=None,
+        help='a regex that sequence titles must match.')
+
+    parser.add_argument(
+        '--negativeTitleRegex', default=None,
+        help='a regex that sequence titles must not match.')
+
+    parser.add_argument(
+        '--truncateTitlesAfter', default=None,
+        help=('a string that titles will be truncated beyond. If the '
+              'truncated version of a title has already been seen, '
+              'that title will be skipped.'))
+
+    parser.add_argument(
+        '--minSequenceLen', type=int, default=None,
+        help='sequences of lesser length will be elided.')
+
+    parser.add_argument(
+        '--maxSequenceLen', type=int, default=None,
+        help='sequences of greater length will be elided.')
+
+    parser.add_argument(
+        '--taxonomy', default=None,
+        help=('a string of the taxonomic group on which should be '
+              'filtered. eg "Vira" will filter on viruses.'))
+
+    args = parser.parse_args()
+
+    # Flatten lists of lists that we get from using both nargs='+' and
+    # action='append'. We use both because it allows people to use (e.g.)
+    # --json on the command line either via "--json file1 --json file2" or
+    # "--json file1 file2", or a combination of these. That way it's not
+    # necessary to remember which way you're supposed to use it and you also
+    # can't be hit by the subtle problem encountered in
+    # https://github.com/acorg/dark-matter/issues/453
+    jsonFiles = list(chain.from_iterable(args.json))
+    whitelist = (
+        set(chain.from_iterable(args.whitelist)) if args.whitelist else None)
+    blacklist = (
+        set(chain.from_iterable(args.blacklist)) if args.blacklist else None)
+
+    # TODO: Add a --readClass command-line option in case we want to
+    # process FASTA containing AA sequences.
+    if args.fasta:
+        reads = FastaReads(list(chain.from_iterable(args.fasta)))
+    else:
+        reads = FastqReads(list(chain.from_iterable(args.fastq)))
+
+    if args.matcher == 'blast':
+        from dark.blast.alignments import BlastReadsAlignments
+        readsAlignments = BlastReadsAlignments(reads, jsonFiles)
+    else:
```

Comment:
This assumes that we only process blast or diamond, right? What if we add another matcher?